### PR TITLE
add test, change log and refine code

### PR DESCRIPTION
**Pull request by lucusfly**
This change is [Reviewable](https://reviewable.io/reviews/seeleteam/go-seele/35)

### CODE REVIEW

File: contract/log.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import (
+	slog "github.com/seeleteam/go-seele/log"
+)
+
+var log *slog.SeeleLog
+
+func init()  {
+	log = slog.GetLogger("contract", true)
+}
```
Comment (WTer) [inline contract/log.go]: take care this warn

File: contract/request.go [hunk 2]
```diff
@@ -25,12 +25,17 @@ func (operator *Operator) Register(code []byte) {
 func (operator *Operator) sendTx(toAddr string, txType int, payload []byte) {
 	hash := crypto.Keccak256Hash(payload)
 
+	sig, err := crypto.NewSignature(operator.privKey, hash)
+	if err != nil {
+		log.Error(err.Error())
```
Comment (WTer) [inline contract/request.go]: if something wrong, do we need to return?

File: p2p/discovery/bucket.go [hunk 3]
```diff
@@ -50,8 +49,8 @@ func (b *bucket) addNode(node *Node) {
 	}
 }
 
-// hasNode check if the bucket already have this node, if so, return its index, otherwise, return -1
-func (b *bucket) hasNode(node *Node) int {
+// findNode check if the bucket already have this node, if so, return its index, otherwise, return -1
+func (b *bucket) findNode(node *Node) int {
 	b.lock.Lock()
```
Comment (WTer) [inline p2p/discovery/bucket.go]: nice to change to read lock for better performance.

File: p2p/discovery/log.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import (
+	slog "github.com/seeleteam/go-seele/log"
+)
+
+var log *slog.SeeleLog
+
+func init()  {
+	log = slog.GetLogger("discovery", true)
+}
```
Comment (WTer) [inline p2p/discovery/log.go]: warn

File: contract/request.go (hunk 2 above)
Comment (lucusfly) [reply contract/request.go]: just keep what it is before, maybe we will refine code later

Comment (lucusfly) [reply contract/request.go]: change to panic

**Comment by lucusfly** [comment]
Review status: 0 of 17 files reviewed at latest revision, 4 unresolved discussions.

---

*[contract/log.go, line 16 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/35#-L5gsk_1NH4Kr44wUtfn-r1-16:-L5gyB5gDenDAga3XEDL:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/491844beece671181287ca5cea1547f4b9515cbf/contract/log.go#L16)):*
<details><summary><i>Previously, WTer wrote…</i></summary><blockquote>

take care this warn
</blockquote></details>

Done.

---

*[contract/request.go, line 30 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/35#-L5gsk_1NH4Kr44wUtfb-r1-30:-L5gyA9MIg7H7LMUDm79:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/491844beece671181287ca5cea1547f4b9515cbf/contract/request.go#L30)):*
<details><summary><i>Previously, lucusfly (lucus) wrote…</i></summary><blockquote>

change to panic
</blockquote></details>

Done.

---

*[p2p/discovery/log.go, line 16 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/35#-L5gsk_1NH4Kr44wUtfo-r1-16:-L5gy-excrw7gRc-UxxY:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/491844beece671181287ca5cea1547f4b9515cbf/p2p/discovery/log.go#L16)):*
<details><summary><i>Previously, WTer wrote…</i></summary><blockquote>

warn
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/seeleteam/go-seele/35)*

**Comment by lucusfly** [comment]
Review status: 0 of 17 files reviewed at latest revision, 4 unresolved discussions.

---

*[p2p/discovery/bucket.go, line 54 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/35#-L5gsk_1NH4Kr44wUtff-r1-54:-L5h9uuKhN7T4VACjQI0:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/491844beece671181287ca5cea1547f4b9515cbf/p2p/discovery/bucket.go#L54)):*
<details><summary><i>Previously, WTer wrote…</i></summary><blockquote>

nice to change to read lock for better performance.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/seeleteam/go-seele/35)*

**Comment by lucusfly** [comment]
Review status: 0 of 17 files reviewed at latest revision, 4 unresolved discussions.

---

*[contract/log.go, line 16 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/35#-L5gsk_1NH4Kr44wUtfn-r1-16:-L5hApyyC6egI-axFPcR:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/491844beece671181287ca5cea1547f4b9515cbf/contract/log.go#L16)):*
<details><summary><i>Previously, lucusfly (lucus) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---

*[contract/request.go, line 30 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/35#-L5gsk_1NH4Kr44wUtfb-r1-30:-L5hAkE4f4Kc8VDPPYky:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/491844beece671181287ca5cea1547f4b9515cbf/contract/request.go#L30)):*
<details><summary><i>Previously, lucusfly (lucus) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---

*[p2p/discovery/bucket.go, line 54 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/35#-L5gsk_1NH4Kr44wUtff-r1-54:-L5hAmNAOUwckZ6BHjo5:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/491844beece671181287ca5cea1547f4b9515cbf/p2p/discovery/bucket.go#L54)):*
<details><summary><i>Previously, lucusfly (lucus) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---

*[p2p/discovery/log.go, line 16 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/35#-L5gsk_1NH4Kr44wUtfo-r1-16:-L5hAnmtIVXvvRPJjoF5:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/491844beece671181287ca5cea1547f4b9515cbf/p2p/discovery/log.go#L16)):*
<details><summary><i>Previously, lucusfly (lucus) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/seeleteam/go-seele/35)*

**Pull request merged**